pkg/lib/packagekit.js

```diff
@@ -404,33 +400,78 @@ export function check_missing_packages(names, progress_cb) {
         }
     }
 
-    return refresh().then(resolve)
+    return refresh(false, progress_cb).then(resolve)
             .then(simulate)
             .then(get_details);
 }
 
-/* Check a list of packages whether they are installed.
+/**
+ * Check a list of packages whether they are installed.
  *
- * This is a lightweight version of check_missing_packages() which does not
- * refresh, simulates, or retrieves details. It just checks which of the given package
- * names are already installed, and returns a Set of the missing ones.
+ * @param {string[]} names - names of packages which should be installed
+ * @return {Promise<boolean>} true if packages are installed
  */
-export function check_uninstalled_packages(names) {
-    const uninstalled = new Set(names);
+export async function is_installed(names, progress_cb) {
+    const installed = [];
+
+    if ((names && names.length === 0))
+        return true;
+
+    await cancellableTransaction("Resolve",
+                                 [Enum.FILTER_ARCH | Enum.FILTER_NOT_SOURCE | Enum.FILTER_INSTALLED, names],
+                                 progress_cb,
+                                 {
+                                     Package: (info, package_id) => {
+                                         installed.push(package_id);
+                                     },
+                                 });
+
+    return installed.length === names.length;
+}
 
-    if (names.length === 0)
-        return Promise.resolve(uninstalled);
+/**
+ * Check a list of packages whether they are available.
+ *
+ * @param {string[]} names - names of packages which should be available in the repositories
+ * @return {Promise<boolean>} true if packages are available
+ */
+export async function is_available(names, progress_cb) {
+    const available = [];
+
+    if ((names && names.length === 0))
+        return true;
+
+    await cancellableTransaction("Resolve",
+                                 [Enum.FILTER_ARCH | Enum.FILTER_NEWEST | Enum.FILTER_NOT_INSTALLED, names],
+                                 progress_cb,
+                                 {
+                                     Package: (info, package_id) => {
+                                         available.push(package_id);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22498-52f3e529-20251111-125657-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)